src/main/java/ch/astorm/jotlmsg/io/FlatEntryListStructure.java

```diff
@@ -0,0 +1,147 @@
+     */
+    public void setFlatEntryStructures(List<T> flatEntries) {
+        this.flatEntryStructures = flatEntries;
+    }
+
+    /**
+     * Adds a FlatEntryStructure.
+     * 
+     * @param flatEntry FlatEntryStructure or subclass to add.
+     */
+    public void addFlatEntryStructure(T flatEntry) {
+        flatEntryStructures.add(flatEntry);
+
+        long totalSize = 0;
+
+        for (FlatEntryStructure fes : flatEntryStructures) {
+            totalSize += fes.getSize();
+            totalSize += 4; // 4 bytes variable size
+
+            // Consider 4 byte alignment
+            totalSize = (totalSize + 4) & ~3;
+        }
+
+        size = totalSize;
+        count = (long) flatEntryStructures.size();
+    }
+
+    /**
+     * Constructor, creates an empty java list representation.
+     * 
+     */
+    public FlatEntryListStructure() {
+    };
+
+    /**
+     * Constructor, creates a java list representation from byte array.
+     * 
+     * @param clazz Class/subclass of {@link ch.astorm.jotlmsg.io.FlatEntryListStructure} used to create list elements.
+     * @param bytes Byte array source.
+     */
+    public FlatEntryListStructure(Class<T> clazz, byte[] bytes) {
+        ByteBuffer bf = ByteBuffer.wrap(bytes).order(ByteOrder.LITTLE_ENDIAN);
+
+        // Count and size are stored in 4 bytes.
+        count = bf.getInt();
+        size = bf.getInt();
+
+        for (int i = 0; i < count; i++) {
+            flatEntryStructures.add(fesf.createFlatEntryStructure(clazz, bf));
+            // Consider 4 byte alignment
+            bf.position((bf.position() + 4) & ~3);
+        }
+    }
+
+    /**
+     * Creates a byte array.
+     * 
+     * @return Byte array representation of this list.
+     */
+    public byte[] toBytes() {
+        // Calculate total size of ByteBuffer
+        long totalSize = 8; // 4 bytes variable count + 4 bytes variable size
+
+        for (FlatEntryStructure fes : flatEntryStructures) {
+            totalSize += fes.getSize();
+            totalSize += 4; // 4 bytes variable size
+
+            // Consider 4 byte alignment
+            totalSize = (totalSize + 4) & ~3;
+        }
+
+        // Allocate the complete ByteBuffer.
+        ByteBuffer bf = ByteBuffer.allocate((int) totalSize).order(ByteOrder.LITTLE_ENDIAN);
+
+        // Store count and size in 4 bytes.
+        bf.putInt((int) flatEntryStructures.size()); // Number of FlatEntryStructures.
+        bf.putInt((int) totalSize - 8); // Number of bytes of all FlatEntryStructures.
+
+        for (FlatEntryStructure fes : flatEntryStructures) {
+            bf.putInt((int) fes.getSize());
+            bf.put(fes.getEntryID());
+
+            // Consider 4 byte alignment
+            bf.position((bf.position() + 4) & ~3);
+        }
+        size = totalSize - 8;
+        return bf.array();
+    }
+
+    /**
+     * Iterator.
+     */
+    public Iterator<T> iterator() {
```

Comment:
Note sure about this javadoc comment :thinking: Maybe just remove it ? Also, it seems that `@Override` is missing.

Reply:
I think so too. The comment is useless and I will add the missing override anotation.